cpu/sam0_common/sam0_sdhc/sdhc.c

```diff
@@ -0,0 +1,1014 @@
+}
+
+static void _reset_all(sdhc_state_t *state)
+{
+    SDHC_DEV->SRR.reg = SDHC_SRR_SWRSTALL;
+    while (SDHC_DEV->SRR.bit.SWRSTALL) {}
+    state->need_init = true;
+    state->error = 0;
+}
+
+static void _init_clocks(sdhc_state_t *state)
+{
+    sam0_gclk_enable(SDHC_CLOCK);
+    sam0_gclk_enable(SDHC_CLOCK_SLOW);
+
+    gpio_init_mux(state->wp, SAM0_SDHC_MUX);
+    gpio_init_mux(state->cd, SAM0_SDHC_MUX);
+
+    if (SDHC_DEV == SDHC0) {
+        /* data pins are fixed */
+        gpio_init_mux(SAM0_SDHC0_PIN_SDDAT0, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC0_PIN_SDDAT1, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC0_PIN_SDDAT2, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC0_PIN_SDDAT3, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC0_PIN_SDCMD, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC0_PIN_SDCK, SAM0_SDHC_MUX);
+
+        GCLK->PCHCTRL[SDHC0_GCLK_ID].reg = GCLK_PCHCTRL_CHEN
+                                         | GCLK_PCHCTRL_GEN(SDHC_CLOCK);
+        GCLK->PCHCTRL[SDHC0_GCLK_ID_SLOW].reg = GCLK_PCHCTRL_CHEN
+                                              | GCLK_PCHCTRL_GEN(SDHC_CLOCK_SLOW);
+        MCLK->AHBMASK.bit.SDHC0_ = 1;
+        isr_ctx_0 = state;
+        NVIC_EnableIRQ(SDHC0_IRQn);
+    }
+
+    if (SDHC_DEV == SDHC1) {
+        /* data pins are fixed */
+        gpio_init_mux(SAM0_SDHC1_PIN_SDDAT0, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC1_PIN_SDDAT1, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC1_PIN_SDDAT2, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC1_PIN_SDDAT3, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC1_PIN_SDCMD, SAM0_SDHC_MUX);
+        gpio_init_mux(SAM0_SDHC1_PIN_SDCK, SAM0_SDHC_MUX);
+
+        GCLK->PCHCTRL[SDHC1_GCLK_ID].reg = GCLK_PCHCTRL_CHEN
+                                         | GCLK_PCHCTRL_GEN(SDHC_CLOCK);
+        GCLK->PCHCTRL[SDHC1_GCLK_ID_SLOW].reg = GCLK_PCHCTRL_CHEN
+                                              | GCLK_PCHCTRL_GEN(SDHC_CLOCK_SLOW);
+        MCLK->AHBMASK.bit.SDHC1_ = 1;
+        isr_ctx_1 = state;
+        NVIC_EnableIRQ(SDHC1_IRQn);
+    }
+}
+
+int sdhc_init(sdhc_state_t *state)
+{
+    bool f8;
+    uint32_t response;
+
+    /* set the initial clock slow, single bit and normal speed */
+    state->type = CARD_TYPE_SD;
+    state->version = CARD_VER_UNKNOWN;
+    state->rca = 0;
+    state->bus_width = 1;
+    state->high_speed = false;
+    state->clock = SDHC_SLOW_CLOCK_HZ;
+    mutex_init(&state->sync);
+    mutex_lock(&state->sync);
+
+    _init_clocks(state);
+    _reset_all(state);
+
+    SDHC_DEV->TCR.reg = 14;                     /* max timeout is 14 or about 1sec */
+    SDHC_DEV->PCR.reg = SDHC_PCR_SDBPWR | SDHC_PCR_SDBVSEL_3V3;
+    SDHC_DEV->NISTER.reg = SDHC_NISTER_MASK;    /* clear all normal interrupt bits */
+    SDHC_DEV->EISTER.reg = SDHC_EISTER_MASK;    /* clear all error interrupt bits  */
+
+    sdhc_set_hc(state);
+    /* 74 startup clocks (190us) */
+    ztimer_sleep(ZTIMER_USEC, 190);
```

Comment:
`ZTIMER_MSEC` dependency was pulled in SAM0 `Makefile.dep` but not `ZTIMER_USEC`

Reply:
ops, I want to be able to use that from riotboot too, so let's just take whatever is available. We only need a *minimum* delay here anyway.